diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ccd9b3..216ba71 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,19 +21,11 @@ else()
   message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++17 support. Please use a different C++ compiler.")
 endif()
 
-set(EIGEN_VERSION 3.3)
-find_package(Eigen3 ${EIGEN_VERSION} QUIET)
-if(NOT EIGEN3_FOUND)
-  set(BUILD_TESTING OFF CACHE INTERNAL "")
-  FetchContent_Declare(eigen
-    GIT_REPOSITORY  https://gitlab.com/libeigen/eigen.git
-    GIT_TAG         ${EIGEN_VERSION}
-    GIT_SHALLOW     ON)
-  FetchContent_MakeAvailable(eigen)
-  unset(BUILD_TESTING CACHE)
-endif()
+#set(EIGEN_VERSION 3.3)
+#find_package(Eigen3 ${EIGEN_VERSION} REQUIRED)
+include_directories(${Eigen3_INCLUDE_DIR})
 
-add_library(${PROJECT_NAME} SHARED
+add_library(${PROJECT_NAME} STATIC
   ${CMAKE_CURRENT_SOURCE_DIR}/src/BYTETracker.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/src/KalmanFilter.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/src/lapjv.cpp
@@ -44,7 +36,9 @@ add_library(${PROJECT_NAME} SHARED
 target_include_directories(${PROJECT_NAME} PUBLIC
   ${CMAKE_CURRENT_SOURCE_DIR}/include
   )
-target_link_libraries(${PROJECT_NAME} Eigen3::Eigen)
+
+install(TARGETS ${PROJECT_NAME})
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/ DESTINATION include)
 
 # Build the tests if the 'BUILD_BYTETRACK_TEST' variable is set to 'ON'
 set(BUILD_BYTETRACK_TEST OFF CACHE BOOL "The flag whether to build the tests or not")
