diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5b99060ee..2ec7fe126 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -282,19 +282,21 @@ if(UNIX AND DNNL_INSTALL_MODE STREQUAL "BUNDLE_V2")
     install(CODE "file(WRITE \"${CMAKE_INSTALL_PREFIX}/${LIB_CONFIG_INSTALL_DIR}/${LIB_PACKAGE_NAME}-targets.cmake\" \"\${TARGETS_CONTENT}\")")
 endif()
 
+# [GML] Looks like these files were removed in openvinotoolkit's fork,
+# [GML] so disable this part.
 # Install custom find modules for transitive dependencies
-if("${DNNL_CPU_THREADING_RUNTIME}" MATCHES "^(TBB|TBB_AUTO)$")
-    if(WIN32)
-        install(FILES "../cmake/win/TBBConfig.cmake" RENAME "FindTBB.cmake"
-            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
-    elseif(APPLE)
-        install(FILES "../cmake/mac/TBBConfig.cmake" RENAME "FindTBB.cmake"
-            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
-    elseif(UNIX)
-        install(FILES "../cmake/lnx/TBBConfig.cmake" RENAME "FindTBB.cmake"
-            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
-    endif()
-endif()
+#if("${DNNL_CPU_THREADING_RUNTIME}" MATCHES "^(TBB|TBB_AUTO)$")
+#    if(WIN32)
+#        install(FILES "../cmake/win/TBBConfig.cmake" RENAME "FindTBB.cmake"
+#            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
+#    elseif(APPLE)
+#        install(FILES "../cmake/mac/TBBConfig.cmake" RENAME "FindTBB.cmake"
+#            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
+#    elseif(UNIX)
+#        install(FILES "../cmake/lnx/TBBConfig.cmake" RENAME "FindTBB.cmake"
+#            DESTINATION ${LIB_CONFIG_INSTALL_DIR})
+#    endif()
+#endif()
 
 if(DNNL_GPU_RUNTIME STREQUAL "OCL")
     install(FILES
@@ -316,3 +318,12 @@ if(WIN32)
     endif()
     set(CTESTCONFIG_PATH "${CTESTCONFIG_PATH}" PARENT_SCOPE)
 endif()
+
+
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+        DESTINATION include
+        FILES_MATCHING
+        PATTERN "*.h"
+        PATTERN "*.hpp")
+
