diff --git a/configure b/configure
index 7b8ce13..93b626f 100755
--- a/configure
+++ b/configure
@@ -3722,16 +3722,16 @@ libdir_default='${prefix}/lib'
 mandir_default='${prefix}/share/man'
 
 # toolchain
-ar_default="ar"
-cc_default="gcc"
-cxx_default="g++"
-host_cc_default="gcc"
+ar_default="$AR"
+cc_default="$CC"
+cxx_default="$CXX"
+host_cc_default="$CC"
 doxygen_default="doxygen"
 install="install"
 ln_s_default="ln -s -f"
-nm_default="nm -g"
+nm_default="$NM -g"
 pkg_config_default=pkg-config
-ranlib_default="ranlib"
+ranlib_default="$RANLIB"
 strip_default="strip"
 version_script='--version-script'
 objformat="elf32"
