diff --git a/mediapipe/framework/port/opencv_video_inc.h b/mediapipe/framework/port/opencv_video_inc.h
index 5f06d923..7a11ba4a 100644
--- a/mediapipe/framework/port/opencv_video_inc.h
+++ b/mediapipe/framework/port/opencv_video_inc.h
@@ -84,16 +84,6 @@ inline int fourcc(char c1, char c2, char c3, char c4) {
 #include <opencv2/video.hpp>
 #include <opencv2/videoio.hpp>
 
-#if CV_VERSION_MAJOR == 4 && !defined(MEDIAPIPE_MOBILE)
-#include <opencv2/optflow.hpp>
-
-namespace cv {
-inline Ptr<DenseOpticalFlow> createOptFlow_DualTVL1() {
-  return optflow::createOptFlow_DualTVL1();
-}
-}  // namespace cv
-#endif
-
 namespace mediapipe {
 inline int fourcc(char c1, char c2, char c3, char c4) {
   return cv::VideoWriter::fourcc(c1, c2, c3, c4);
