diff --git a/mediapipe/examples/desktop/hand_tracking/BUILD b/mediapipe/examples/desktop/hand_tracking/BUILD
index f19fa5a9..2c17d012 100644
--- a/mediapipe/examples/desktop/hand_tracking/BUILD
+++ b/mediapipe/examples/desktop/hand_tracking/BUILD
@@ -14,7 +14,7 @@
 
 licenses(["notice"])
 
-package(default_visibility = ["//mediapipe/examples:__subpackages__"])
+package(default_visibility = ["//visibility:public"])
 
 cc_binary(
     name = "hand_tracking_tflite",
