diff --git a/foreign_cc/private/cmake_script.bzl b/foreign_cc/private/cmake_script.bzl
index bfe21bd..6ca0033 100644
--- a/foreign_cc/private/cmake_script.bzl
+++ b/foreign_cc/private/cmake_script.bzl
@@ -120,9 +120,9 @@ def create_cmake_script(
     # by setting CMAKE_SYSTEM_NAME and CMAKE_SYSTEM_PROCESSOR,
     # see https://github.com/bazelbuild/rules_foreign_cc/issues/289,
     # and https://github.com/bazelbuild/rules_foreign_cc/pull/1062
-    if target_os != host_os and target_os != "unknown":
-        params.cache.update(_TARGET_OS_PARAMS.get(target_os, {}))
-        params.cache.update(_TARGET_ARCH_PARAMS.get(target_arch, {}))
+    # GML: building cmake under bazel should always be treated as cross-compilation.
+    params.cache.update(_TARGET_OS_PARAMS.get(target_os, {}))
+    params.cache.update(_TARGET_ARCH_PARAMS.get(target_arch, {}))
 
     set_env_vars = [
         "export {}=\"{}\"".format(key, _escape_dquote_bash(params.env[key]))
