diff --git a/foreign_cc/cmake.bzl b/foreign_cc/cmake.bzl
index 6f08f44..56bd786 100644
--- a/foreign_cc/cmake.bzl
+++ b/foreign_cc/cmake.bzl
@@ -263,7 +263,7 @@ def _create_configure_script(configureParameters):
         install_prefix = "$$INSTALLDIR$$",
         root = root,
         no_toolchain_file = no_toolchain_file,
-        user_cache = dict(ctx.attr.cache_entries),
+        user_cache = expand_locations_and_make_variables(ctx, ctx.attr.cache_entries, "cache_entries", data),
         user_env = expand_locations_and_make_variables(ctx, ctx.attr.env, "env", data),
         options = expand_locations_and_make_variables(ctx, ctx.attr.generate_args, "generate_args", data),
         cmake_commands = cmake_commands,
