diff --git a/aspects.bzl b/aspects.bzl
index 616f0ee..7de5a48 100644
--- a/aspects.bzl
+++ b/aspects.bzl
@@ -91,9 +91,11 @@ def _get_compile_flags(dep):
     options = []
     compilation_context = dep[CcInfo].compilation_context
     for define in compilation_context.defines.to_list():
+        define = define.replace('"', '\\"')
         options.append("-D\"{}\"".format(define))
 
     for define in compilation_context.local_defines.to_list():
+        define = define.replace('"', '\\"')
         options.append("-D\"{}\"".format(define))
 
     for system_include in compilation_context.system_includes.to_list():
diff --git a/generate.py b/generate.py
index b7764b1..32f08a5 100755
--- a/generate.py
+++ b/generate.py
@@ -82,8 +82,12 @@ if __name__ == "__main__":
     query = ('kind("^cc_(library|binary|test|inc_library|proto_library)", {query_expr}) ' +
              'union kind("^objc_(library|binary|test)", {query_expr})').format(
                  query_expr=args.query_expr)
-    query_cmd = [_BAZEL, 'query']
-    query_cmd.extend(['--noshow_progress', '--noshow_loading_progress', '--output=label'])
+    query_cmd = [_BAZEL, 'cquery']
+    query_cmd.extend(['--noshow_progress', '--noshow_loading_progress'])
+    query_cmd.extend([
+        "--output=starlark",
+        "--starlark:expr='' if providers(target) and 'IncompatiblePlatformProvider' in providers(target) else target.label",
+    ])
     query_cmd.append(query)
 
     targets_file = tempfile.NamedTemporaryFile()
